python/cugraph/cugraph/sampling/random_walks.py

```diff
@@ -126,13 +128,39 @@ def random_walks(
 
     """
 
-    if legacy_result_type:
+    warning_msg = (
+            "random_walks is deprecated and will be removed "
+            "in the next release in favor of uniform_random_walks"
+        )
+    warnings.warn(warning_msg, FutureWarning)
+
+    # FIXME: Coalesced path results have been deprecated and should no longer be
+    # supported in 25.02.
+    # Context for legacy_result_type: The initial implementation of random_walks
+    # returned results where the vertex and weight path are proportional to the
+    # number of vertices instead of the number of edges hence the flag
+    # 'legacy_result_type' was created. This flag should be removed in favor of
+    # returning results paths proprtional to the number of edges. Furthermore,
+    # Coalesced path results should also be removed in favor of always returning
+    # padded results. The flags 'legacy_result_type' and 'use_padding" should be
+    # removed. 
+
+    if legacy_result_type or use_padding is False:
```

Comment:
Note... it seems like the old mechanism (coalesced format) is not particularly helpful other than a modest savings in memory on a small result type anyway.

We don't intend to support that in the C++ code any longer.  If we need to continue to support this old mechanism while it is deprecated before we can delete it I would suggest implementing something to convert the format within the python layer.

Reply:
We still have the old mechanism around and I deprecated the coalesced path in the python API. Since there is no real benefit of using the coalesced path, I suggest to not support in the new API. @rlratzel any thoughts?